### PR TITLE
Remove in-chat onboarding buttons for new chats

### PR DESCRIPTION
**Pull request by toolmantim**
Show the in-chat onboarding buttons only for first time users who haven't created a new chat.

| Before | After |
| - | - |
| Screenshot 2023-07-31 at 4 01 19 pm | Screenshot 2023-07-31 at 4 01 38 pm |

## Test plan

- Clear history and view chat. Chat should have welcome buttons
- Create a new chat. Chat should not have welcome buttons
- Navigate chats with history, verified they don't have the welcome buttons

### CODE REVIEW

File: vscode/webviews/App.tsx [hunk 1]
```diff
@@ -179,6 +179,7 @@ export const App: React.FunctionComponent<{ vscodeAPI: VSCodeWrapper }> = ({ vsc
                             pluginsDevMode={Boolean(config?.pluginsDebugEnabled)}
                             setSuggestions={setSuggestions}
                             telemetryService={telemetryService}
+                            isFirstChat={false /* TODO */}
```
Comment (toolmantim) [inline vscode/webviews/App.tsx:182]: I tried to figure out how to calc this value from the data structures available, but even live debugging failed me. Would appreciate some eng help on this 🙏🏼

**Comment by umpox** [comment]
@toolmantim Should this only show on the users first ever chat?

If so, I wonder if there's much value in keeping this functionality as it is. I'd imagine most users might never use this, as they could just skip past that first chat and never see it again. Understand that we might want to do more here in future for onboarding though.

Other question: If a user clears their history, should this then go to their oldest chat, or disappear completely?

**Comment by toolmantim** [comment]
@umpox yeah good idea, it would be okay just to hide these as soon as someone uses the "New Chat" button! I'll look into that.

I would like to keep the buttons in general for now to ensure we don't tank our onboarding metrics without properly testing and looking at existing numbers. But removing them in the future as we evolve things probably makes sense!

**Review by abeatrix — APPROVED** [review]

**Comment by chenkc805** [comment]
I'm seeing a drop in retention, and I have a hunch that this change is the cause: https://sourcegraph.slack.com/archives/C05AGQYD528/p1691475546298529

@toolmantim @umpox does this affect brand new installs?

**Pull request merged**